### PR TITLE
Fix code in RST files

### PR DESCRIPTION
**Pull request by Bultako**
This PR fixes Python code in present RST files so to provide functional code working in the documentation.
-  In some cases the code has been removed and links added to tutorial notebooks illustrating how to use the API.
- Moving notebooks to API docs should come in another PR. 
- The tests have been run with command `pytest --doctest-glob="*.rst" docs/` 
- These tests have been enabled also in the CI github workflows.

### CODE REVIEW

**Review by adonath — COMMENTED** [review]
Thanks @Bultako , for now I left two minor comments... +1 to make this part of our CI!

File: docs/analysis/index.rst [hunk 1]
```diff
@@ -20,193 +20,6 @@ IPython sessions. The high-level user interface could also be used to automatise
 processes driven by parameters declared in a configuration file in YAML format
 that addresses the most common analysis use cases identified.
 
-.. _analysis_start:
```
[the remark this answers is not in the document]
Comment (Bultako) [reply docs/analysis/index.rst:23]: Yes, moving notebooks to API docs will come in another PR.

File: docs/data/index.rst [hunk 2]
```diff
@@ -25,14 +25,6 @@ You can use the `~gammapy.data.EventList` class to load IACT gamma-ray event lis
     >>> filename = '$GAMMAPY_DATA/hess-dl3-dr1/data/hess_dl3_dr1_obs_id_023523.fits.gz'
     >>> events = EventList.read(filename)
 
-To load Fermi-LAT event lists, use the `~gammapy.data.EventListLAT` class:
-
-.. code-block:: python
```
[the remark this answers is not in the document]
Comment (Bultako) [reply docs/data/index.rst:30]: ok, done.

**Review by adonath — APPROVED** [review]
Thanks @Bultako, no further comments from my side...

**Pull request merged**